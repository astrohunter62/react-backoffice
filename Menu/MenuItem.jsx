import React from 'react'
import PropTypes from 'prop-types'

import { ListItem, ListItemIcon, ListItemText } from 'material-ui/List'

const MenuItem = ({ redirectTo, url, title, icon: Icon }) => {
  return (
    <ListItem button onClick={() => redirectTo(url)}>
      {Icon? (
        <ListItemIcon>
          <Icon />
        </ListItemIcon>
      ) : null}
      <ListItemText primary={title} />
    </ListItem>
  )
}

MenuItem.propTypes = {
  redirectTo: PropTypes.func.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  icon: PropTypes.func,
}

export default MenuItem
