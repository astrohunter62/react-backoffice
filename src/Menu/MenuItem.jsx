import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import withStyles from 'material-ui/styles/withStyles'
import { ListItem, ListItemIcon, ListItemText } from 'material-ui/List'

const styles = {
  disabled: {
    opacity: 0.5,
    pointerEvents: 'none',
  },
}

const MenuItem = ({
  redirectTo,
  url,
  title,
  icon: Icon,
  disabled,
  classes,
}) => (
  <ListItem
    button
    onClick={() => redirectTo(url)}
    className={classNames({
        [classes.disabled]: disabled,
      })}
  >
    {Icon ? (
      <ListItemIcon>
        <Icon />
      </ListItemIcon>
      ) : null}
    <ListItemText primary={title} />
  </ListItem>
)

MenuItem.propTypes = {
  redirectTo: PropTypes.func.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  icon: PropTypes.func,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

MenuItem.defaultProps = {
  disabled: false,
  icon: null,
}

export default withStyles(styles)(MenuItem)
