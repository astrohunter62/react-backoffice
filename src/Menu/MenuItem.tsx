import React, { FunctionComponent } from "react";
import classNames from "classnames";
import {
  ListItem,
  ListItemIcon,
  ListItemText,
  makeStyles,
} from "@material-ui/core";

const useStyles = makeStyles(() => ({
  isDisabled: {
    opacity: 0.5,
    pointerEvents: "none",
  },
}));

type MenuItemProps = {
  redirectTo: (...args: any[]) => any;
  url: string;
  title: string;
  isDisabled?: boolean;
  icon?: JSX.Element;
  className?: string;
};

const MenuItem: FunctionComponent<MenuItemProps> = ({
  redirectTo,
  url,
  title,
  icon,
  isDisabled = false,
  className,
}) => {
  const classes = useStyles();

  return (
    <ListItem
      button
      onClick={() => redirectTo(url)}
      className={classNames(className, {
        [classes.isDisabled]: isDisabled,
      })}
    >
      {icon && <ListItemIcon>{icon}</ListItemIcon>}

      <ListItemText primary={title} />
    </ListItem>
  );
};

export default MenuItem;
