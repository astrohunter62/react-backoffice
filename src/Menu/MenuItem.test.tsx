import React from "react";
import { shallow } from "enzyme";

import Icon from "@material-ui/icons/AccessAlarm";

import MenuItem from "./MenuItem";

it("renders correctly", () => {
  const tree = shallow(
    <MenuItem title="Title" url="/" redirectTo={() => {}} />,
  );

  expect(tree).toMatchSnapshot();
});

it("renders correctly with icon", () => {
  const tree = shallow(
    <MenuItem title="Title" url="/" redirectTo={() => {}} icon={Icon} />,
  );

  expect(tree).toMatchSnapshot();
});

it("renders correctly disabled", () => {
  const tree = shallow(
    <MenuItem title="Title" url="/" redirectTo={() => {}} isDisabled />,
  );

  expect(tree).toMatchSnapshot();
});
