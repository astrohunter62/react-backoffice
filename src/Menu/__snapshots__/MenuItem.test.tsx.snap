// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<WithStyles(ForwardRef(ListItem))
  button={true}
  className=""
  onClick={[Function]}
>
  <WithStyles(ForwardRef(ListItemText))
    primary="Title"
  />
</WithStyles(ForwardRef(ListItem))>
`;

exports[`renders correctly disabled 1`] = `
<WithStyles(ForwardRef(ListItem))
  button={true}
  className="makeStyles-isDisabled-1"
  onClick={[Function]}
>
  <WithStyles(ForwardRef(ListItemText))
    primary="Title"
  />
</WithStyles(ForwardRef(ListItem))>
`;

exports[`renders correctly with icon 1`] = `
<WithStyles(ForwardRef(ListItem))
  button={true}
  className=""
  onClick={[Function]}
>
  <WithStyles(ForwardRef(ListItemIcon))>
    <AccessAlarmIcon />
  </WithStyles(ForwardRef(ListItemIcon))>
  <WithStyles(ForwardRef(ListItemText))
    primary="Title"
  />
</WithStyles(ForwardRef(ListItem))>
`;
